sync-core/src/test/java/com/cloudant/sync/query/AbstractQueryTestBase.java

```diff
@@ -105,13 +132,38 @@ public void setUp() throws SQLException {
             Assert.fail("Failed to create document revision");
         }
 
-        assertThat(im.ensureIndexed(Arrays.<Object>asList("name", "age"), "basic"), is("basic"));
-        assertThat(im.ensureIndexed(Arrays.<Object>asList("name", "pet"), "pet"), is("pet"));
+    }
+
+    @After
+    public void tearDown() {
+        im.close();
+        assertThat(im.getQueue().isShutdown(), is(true));
+        ds.close();
+        TestUtils.deleteDatabaseQuietly(db);
+        TestUtils.deleteTempTestingDir(factoryPath);
+
+        im = null;
+        ds = null;
+        factory = null;
+        factoryPath = null;
     }
 
     @Test
     public void returnsNullForNoQuery() {
-        Assert.assertNull(im.find(null));
+        assertThat(im.find(null), is(nullValue()));
+    }
+
+    @Test
+    public void returnsNullWhenQueryContainsInvalidValue() {
+        // query - { "name" : { "eq" : "mike" }, "age" : { "$eq" : 12.0f } }
+        Map<String, Object> nameOperator = new HashMap<String, Object>();
+        nameOperator.put("$eq", "mike");
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", nameOperator);
+        Map<String, Object> ageOperator = new HashMap<String, Object>();
+        ageOperator.put("$eq", 12.0f);
```

Comment:
Added a comment to the test description.
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-1a5092ac6698384ba572368c278436fdR157